File: Civitas-Java/Civitas/src/main/java/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 * @author elena
 * @date 29/11/2019
 */
public class CasillaCalle extends Casilla{
    private TituloPropiedad tituloPropiedad = null;
    
    CasillaCalle(TituloPropiedad titulo){ 
        super(titulo.getNombre());
        tituloPropiedad = titulo;
    }
    
    TituloPropiedad getTituloPropiedad(){ return this.tituloPropiedad; }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(iactual, todos)){
           super.informe(iactual, todos);
           Jugador jugador = todos.get(iactual);
            
           if(!tituloPropiedad.tienePropietario())
                jugador.puedeComprarCasilla();
           else
                tituloPropiedad.tramitarAlquiler(jugador);
        }
    } 
    
    @Override
    public String toString(){
        return "Nombre: " + this.getTituloPropiedad().getNombre() +
               "\nTipo Casilla: CALLE";
    }
}

File: Civitas-Ruby/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "mazo_sorpresas"
require_relative "titulo_propiedad"
require_relative "sorpresa"
require_relative "tipo_casilla"
require_relative "tipo_sorpresa"
require_relative "jugador"

module Civitas
  class CasillaCalle < Casilla
    attr_reader :titulo
    public :titulo

    def initialize(titulo)
        super
        @titulo = titulo
    end
    
    @Override
    def recibejugador(iactual, todos)
      if jugadorcorrecto(iactual, todos)
        informe(iactual, todos)
        jugador = todos[iactual]
        
        if @titulo.tienepropietario
          jugador.puedecomprarcasilla
        else
          @titulo.tramitaralquiler(jugador)
        end
      end
    end
    
    @Override
    def to_s
      '\nNombre: ' + @titulo.nombre.to_s +
      '\nTipo Casilla: CALLE'       
    end

      public :titulo
  end
end
